File: DS-Solutions/Python_Solutions/bulb_switch.py
bulbs=[0,1,0,1]
sum=0
count=0
for i in bulbs:
    sum+=i
if sum==len(bulbs):
    print(0)
else:
    for i in range(0,len(bulbs)):
        if bulbs[i]==0:
            bulbs[i+1:]=not(bulbs[i+1:])
            count+=1
        else:
            continue
print(bulbs)

File: DS-Solutions/Cpp_Solutions/bulb_switch.cpp
#include<bits/stdc++.h>
using namespace std;
#define flip(a) (a==0?1:0)

int main()
{
	int bulbs[]={0,1,0,1};
	int size=*(&bulbs+1)-bulbs;
	int sum=0;
	int count=0;
	for(int i=0;i<size;i++)
		sum+=bulbs[i];
	if (sum==size)
		cout<<0;
	else
	{
		for(int i=0;i<size;i++)
		{
			if(bulbs[i]==0)
			{
				for(int j=i;j<size;j++)
					bulbs[j]=flip(bulbs[j]);
				count+=1;
			}
			else 
				continue;			
		}
		cout<<count<<endl;
	}	
}
